src/main.rs

```diff
@@ -109,42 +104,52 @@ fn dropfile_filename(filename: &str) -> String {
 }
 
 fn write_new_door32sys_dropfile(
-    orig_contents: &str, out_path: &Path, socket_fd: i64) -> Result<String, String>
-{
+    orig_contents: &str,
+    out_path: &Path,
+    socket_fd: u64,
+) -> Result<String, Error> {
     if !out_path.is_dir() {
-        return Err(format!("{} is not a directory", out_path.to_string_lossy()));
+        return Err(Error::new(
+            ErrorKind::NotFound,
+            format!("{} is not a directory", out_path.to_string_lossy()),
+        ));
     }
 
-    let dropfile_path = out_path.join(dropfile_filename(DOOR32_SYS_FILENAME));
-
     //
     //  First two lines are as follows:
     //  1 - Comm type (2=telnet)
     //  2 - Shared socket fd
     //  ...the rest is just copied over from the original.
-    //      
-    let mut contents = format!("2\r\n{}\r\n", socket_fd);
-    let remaining_lines = orig_contents.lines().skip(2);
-    for line in remaining_lines {
-        contents.push_str(&format!("{}\r\n", line));
+    //
+    let dropfile_path = out_path.join(dropfile_filename(DOOR32_SYS_FILENAME));
+    let mut file = fs::File::create(&dropfile_path)?;
+    file.write_all(b"2\r\n")?;
```

Comment:
What's the advantage here over the format macro?

Reply:
The main advantage here is that you're not creating unnecessary intermediate values on the heap that are dropped almost immediately. When you, `write!(file, "2\r\n{}\r\n");` this gets expanded into a something that calls `format!()` (or an equivalent) that does a heap allocation to create a string to write into; the allocated memory is almost immediately `Drop`'ed after the write!() invocation. This way avoids the allocations.